xls/tools/run_timing_characterization.py

```diff
@@ -0,0 +1,178 @@
+        config.client_extra_args = '--v 3'
+    else:
+        config.server_extra_args = ''
+        config.client_extra_args = ''
+
+
+def _do_config_kAsap7(config: WorkerConfig):
+    # Clearly there are better data structures to use...
+    asap7_path = f'{config.openroad_path}/flow/platforms/asap7/lib'
+    # OpenROAD merge cells into one lib for yosys - lets use that
+    config.synthesis_libraries = f'{config.openroad_path}/flow/objects/asap7/ibex/base/lib/merged.lib'
+    config.sta_libraries = f'\
+    {asap7_path}/asap7sc7p5t_AO_RVT_FF_nldm_211120.lib.gz \
+    {asap7_path}/asap7sc7p5t_INVBUF_RVT_FF_nldm_220122.lib.gz \
+    {asap7_path}/asap7sc7p5t_OA_RVT_FF_nldm_211120.lib.gz \
+    {asap7_path}/asap7sc7p5t_SIMPLE_RVT_FF_nldm_211120.lib.gz \
+    {asap7_path}/asap7sc7p5t_SEQ_RVT_FF_nldm_220123.lib'
+    config.client_args = '--max_width=8 --min_freq_mhz=1000 --max_freq_mhz=25000'
+
+
+def _do_config_kNangate45(config: WorkerConfig):
+    nangate45_path = f'{config.openroad_path}/flow/platforms/nangate45/lib'
+    config.synthesis_libraries = f'{nangate45_path}/NangateOpenCellLibrary_typical.lib'
+    config.sta_libraries = f'{nangate45_path}/NangateOpenCellLibrary_typical.lib'
+    config.client_args = '--max_width=8 --min_freq_mhz=1000 --max_freq_mhz=11000'
+
+
+def _do_config_kSky130(config: WorkerConfig):
+    sky130_path = f'{config.openroad_path}/flow/platforms/sky130hd/lib'
+    config.synthesis_libraries = f'{sky130_path}/sky130_fd_sc_hd__tt_025C_1v80.lib'
+    config.sta_libraries = f'{sky130_path}/sky130_fd_sc_hd__tt_025C_1v80.lib'
+    config.client_args = '--max_width=8 --min_freq_mhz=1000 --max_freq_mhz=11000'
+
+
+def _do_worker_task(config: WorkerConfig):
+    logging.info('Running')
+    logging.info(f'  Using OpenROAD directory: {config.openroad_path}')
+    logging.info(f'  Using Yosys : {config.yosys_bin}')
+    logging.info(f'  Using STA   : {config.sta_bin}')
+    logging.info(f'  With  target: {config.target}')
+
+    server = f'{config.bazel_path}/xls/synthesis/yosys/yosys_sta_server_main \
+    --yosys_path={config.yosys_bin} \
+    --synthesis_libraries="{config.synthesis_libraries}" \
+    --sta_path={config.sta_bin} \
+    --sta_libraries="{config.sta_libraries}" \
+    --port={config.rpc_port} \
+    {config.server_extra_args}'
+
+    client = f'{config.bazel_path}/xls/synthesis/timing_characterization_client_main \
+    --checkpoint_path {config.client_checkpoint_file} \
+    {config.client_args} {config.client_extra_args} \
+    --port={config.rpc_port}'\
+
+    # create a checkpoint file if not allready there
+    with open(config.client_checkpoint_file, 'w') as f:
+        f.write('')
+
+    start = datetime.datetime.now()
+
+    # start non-blocking process
+    # server_process=subprocess.Popen(server, stdout=subprocess.PIPE,shell=True,text=True)
+    server_process = subprocess.Popen(
+        server, stdout=subprocess.PIPE, shell=True)
```

Comment:
Nit: avoid using shell=True. Instead split the args into separate string elements of a list. This avoids potential unexpected behavior due to shell token splitting and expansion.